### PR TITLE
feat(bindings): update FilterPolicy to match AWS API

### PR DESCRIPTION
**Pull request by Gadam8**
## About the PR

Currently, the FilterPolicy configuration does not match what is outlined in the AWS documentation and API - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-subscription.html. The FilterPolicy should only contain the JSON policy, so no need for the attributes property; which in itself was confusing as the policy can apply to the message attributes or body.

Further to this, FilterPolicyScope was missing which is needed to determine whether the user wishes to filter on the message attributes or the body. I have a PR raised in the AsyncAPI bindings repo to make the changes to the binding documentation - https://github.com/asyncapi/bindings/pull/215. This has been approved and is awaiting merge.

### Changelog

- Add: Added FilterPolicyScope to the Consumer object on the SNS operation binding.
- Add: Adjusted FilterPolicy to remove `attributes` field and make the object the direct input for the policy

### CODE REVIEW

**Comment by VisualBean** [comment]
@dpwdec for reference

**Review by VisualBean — APPROVED** [review]
Looks good. As long as @dpwdec is happy, and this follows the Binding - I dont think there are any problems.

File: test/LEGO.AsyncAPI.Tests/Bindings/Sns/SnsBindings_Should.cs
Comment (VisualBean) [inline test/LEGO.AsyncAPI.Tests/Bindings/Sns/SnsBindings_Should.cs:181]: looks like the extension was removed.
Lets add an extension to test that the binding is extensible

Comment (Gadam8) [reply test/LEGO.AsyncAPI.Tests/Bindings/Sns/SnsBindings_Should.cs:181]: I removed this extension as the `attributes` property no longer exists and the filter policy lives directly under the object now. So I thought the extension for the filter policy object should now be covered by the general consumer extension (https://github.com/LEGO/AsyncAPI.NET/pull/128/files#diff-0b77af8e82f20917a60bde7519aebfcb0c7f67315a86a89636c04ba6b16fd315L204)?

Comment (VisualBean) [reply test/LEGO.AsyncAPI.Tests/Bindings/Sns/SnsBindings_Should.cs:181]: You are absolutely right. I totally missed that!

**Comment by dpwdec** [comment]
@Gadam8 LGTM! I've approved the binding on the AsyncAPI side, just a case of waiting for lead maintainers to approve now. 

@VisualBean Do you think we should leave this open until a merge goes through on the bindings project? Or can we merge this change now in anticipation of the correction on the bindings side?

**Comment by Gadam8** [comment]
Binding PR has now been merged https://github.com/asyncapi/bindings/pull/215

**Review by VisualBean — APPROVED** [review]

**Pull request merged**
